main.py

```diff
@@ -113,39 +113,46 @@ def get_randomizer_settings_v2(seed_id):
 @app.route('/randomizer_settings', methods=['POST'])
 @limiter.limit("10/hour")
 def post_randomizer_settings():
-    seed_dict = request.get_json()
+    seed_request = request.get_json()
     
     try:
-        SeedRequestSchema().load(seed_dict)
+        SeedRequestSchema().load(seed_request)
     except ValidationError as err:
         print(err)
         return err.messages, 400
 
     unique_seed_id = get_unique_seedID(db, firestore_seeds_collection)
-    seed_dict["SeedID"] = unique_seed_id
-    seed = Seed(**seed_dict)
+    seed_request["SeedID"] = unique_seed_id
+    seed_request["SeedValue"] = random.randint(0, 0xFFFFFFFF)    
 
     world_graph = init_world_graph()
 
-    print(f'Request settings {seed.__dict__}')
+    print(f'Request settings {seed_request}')
 
     try:
-        rando_result = web_randomizer(json.dumps(seed.__dict__, default = lambda o: f"<<non-serializable: {type(o).__qualname__}>>"), world_graph)
-    except AssertionError as err:
-        print(err)
-        return str(err), 400
+        rando_result = web_randomizer(json.dumps(seed_request, default = lambda o: f"<<non-serializable: {type(o).__qualname__}>>"), world_graph)
     except Exception as err:
         print(err)
-        db.collection(firestore_failure_collection).document(str(unique_seed_id)).set(seed.__dict__)
+        db.collection(firestore_failure_collection).document(str(unique_seed_id)).set(seed_request)
         raise err
-    
-    seed.PaletteOffset = rando_result.palette_offset
-    seed.CosmeticsOffset = rando_result.cosmetics_offset
-    seed.AudioOffset = rando_result.audio_offset
-    seed.MusicOffset = rando_result.music_offset
-    seed.SeedHashItems = rando_result.hash_items
 
-    db.collection(firestore_seeds_collection).document(str(unique_seed_id)).set(seed.__dict__)
+    # Transfer back the settings model data from the generator
+    seed_result = rando_result.web_settings
+    seed_result["PaletteOffset"] = rando_result.palette_offset
+    seed_result["CosmeticsOffset"] = rando_result.cosmetics_offset
+    seed_result["AudioOffset"] = rando_result.audio_offset
+    seed_result["MusicOffset"] = rando_result.music_offset
+    seed_result["SeedHashItems"] = rando_result.hash_items
+
+    seed_result["SeedID"] = seed_request["SeedID"]
+    seed_result["CreationDate"] = datetime.now(timezone.utc)
+    seed_result["StarRodModVersion"] = seed_request["StarRodModVersion"]
+    seed_result["SettingsString"] = seed_request["SettingsString"]
+    if seed_request.get("WriteSpoilerLog") and seed_request.get("RevealLogInHours") != 0:
```

Comment:
Could exchange the `.get(...)`s with `[...]`, since we expect these to be `not None`.
-> `if seed_request["WriteSpoilerLog"] and seed_request["RevealLogInHours"] != 0:`

Reply:
I did the change but actually I think we should not. If we want to support yaml generation, we should assume every field from seed_request might be null, I'm gonna revert it.

Reply:
Reverted and actually made more fields optional so we don't have yamls crashing from missing some web field

Reply:
Even with an empty yaml all settings fields should be set by the seed generator and returned via the WebSeedResponse.
The only fields we need to look at should be the ones exclusively used by the web generator, so in the case of line 151 it's `RevealLogInHours`, but that one is checked by the SeedRequestSchema beforehand

Reply:
Yeah well I just made it optional in the schema to allow for seed generation without it, it makes sense